app/src/main/java/com/quarkworks/apartmentgroceries/service/SyncGroceryItem.java

```diff
@@ -30,6 +39,7 @@ public static Promise getAll() {
             @Override
             public void done(@Nullable JSONObject jsonObject) {
 
```

Comment:
The JSONObject you receive is nullable, so you need to do a null check before trying to parse it. Ex) 

if(jsonObject == null) {
     Log.e(TAG, "Error getting get grocery items from server");
     promise.onFailure();
     return;
}